cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/domain/Staging.java

```diff
@@ -1,201 +1,66 @@
-        public StagingBuilder buildpackUrl(String buildpackUrl) {
-            this.buildpackUrl = buildpackUrl;
-            return this;
-        }
+import org.cloudfoundry.client.lib.domain.annotation.Nullable;
+import org.immutables.value.Value;
 
-        /*
-         * @param detectedBuildpack raw, free-form information regarding a detected buildpack. It is a read-only property, and should not be
-         * set except when parsing a response. May be null.
-         */
-        public StagingBuilder detectedBuildpack(String detectedBuildpack) {
-            this.detectedBuildpack = detectedBuildpack;
-            return this;
-        }
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 
-        // @param stack the stack to use when staging the application; may be null
-        public StagingBuilder stack(String stack) {
-            this.stack = stack;
-            return this;
-        }
-
-        // @param healthCheckTimeout the amount of time the platform should wait when verifying that an app started; may be null
-        public StagingBuilder healthCheckTimeout(Integer healthCheckTimeout) {
-            this.healthCheckTimeout = healthCheckTimeout;
-            return this;
-        }
-
-        // @param healthCheckType; may be null
-        public StagingBuilder healthCheckType(String healthCheckType) {
-            this.healthCheckType = healthCheckType;
-            return this;
-        }
-
-        // @param healthCheckHttpEndpoint; may be null
-        public StagingBuilder healthCheckHttpEndpoint(String healthCheckHttpEndpoint) {
-            this.healthCheckHttpEndpoint = healthCheckHttpEndpoint;
-            return this;
-        }
-
-        // @param sshEnabled boolean value which shows if the ssh for the app is enabled or disabled; may be null
-        public StagingBuilder sshEnabled(Boolean sshEnabled) {
-            this.sshEnabled = sshEnabled;
-            return this;
-        }
-
-        // @param dockerInfo; may be null if a docker image is not used
-        public StagingBuilder dockerInfo(DockerInfo dockerInfo) {
-            this.dockerInfo = dockerInfo;
-            return this;
-        }
-
-        public Staging build() {
-            return new Staging(this);
-        }
-
-    }
+@Value.Immutable
+@JsonSerialize(as = ImmutableStaging.class)
+@JsonDeserialize(as = ImmutableStaging.class)
+public interface Staging {
 
     /**
      * @return The buildpack url, or null to use the default buildpack detected based on application content
      */
-    public String getBuildpackUrl() {
-        return buildpackUrl;
-    }
+    @Nullable
+    String getBuildpackUrl();
 
     /**
      * @return The start command to use
      */
-    public String getCommand() {
-        return command;
-    }
+    @Nullable
+    String getCommand();
```

Comment:
Yes, a task without a command is like an application without a binary, but even worse, because you can attach a binary to an application at a later point, but the creation of a task resource starts the execution of the task. If we don't supply a command, there'll be an error.